components/src/payments/confirmation.ts

```diff
@@ -0,0 +1,756 @@
+      padding: 16px;
+      font-weight: 600;
+      font-size: 16px;
+      cursor: pointer;
+      transition: all 0.2s ease;
+    }
+
+    .confirm-button:hover:not(:disabled) {
+      background: #059669;
+    }
+
+    .confirm-button:disabled {
+      background: #d1d5db;
+      cursor: not-allowed;
+    }
+
+    .loading-state {
+      display: flex;
+      align-items: center;
+      justify-content: center;
+      padding: 40px;
+      color: var(--text-color);
+      opacity: 0.6;
+    }
+
+    .spinner {
+      width: 20px;
+      height: 20px;
+      border: 2px solid #e5e7eb;
+      border-top: 2px solid var(--primary-color);
+      border-radius: 50%;
+      animation: spin 1s linear infinite;
+      margin-right: 12px;
+    }
+
+    @keyframes spin {
+      0% {
+        transform: rotate(0deg);
+      }
+      100% {
+        transform: rotate(360deg);
+      }
+    }
+
+    .empty-state {
+      text-align: center;
+      color: var(--text-color);
+      opacity: 0.6;
+      padding: 40px 20px;
+    }
+
+    .empty-state-title {
+      font-size: 1.125rem;
+      font-weight: 600;
+      margin-bottom: 8px;
+    }
+
+    .empty-state-description {
+      font-size: 0.875rem;
+    }
+
+    .back-button {
+      background: transparent;
+      border: 2px solid var(--primary-color);
+      color: var(--primary-color);
+      padding: 8px 16px;
+      border-radius: 6px;
+      font-size: 14px;
+      cursor: pointer;
+      transition: all 0.2s ease;
+      margin: 16px;
+      align-self: flex-start;
+    }
+
+    .back-button:hover {
+      background: var(--primary-color);
+      color: white;
+    }
+
+    .payment-form {
+      display: flex;
+      flex-direction: column;
+      gap: 16px;
+      margin-top: 16px;
+    }
+
+    .form-group {
+      display: flex;
+      flex-direction: column;
+    }
+
+    .form-label {
+      display: block;
+      margin-bottom: 8px;
+      font-weight: 500;
+      color: var(--text-color);
+      font-size: 0.875rem;
+    }
+
+    .form-input {
+      width: 100%;
+      padding: 12px 16px;
+      border: 1px solid #d1d5db;
+      border-radius: 6px;
+      font-size: 16px;
+      box-sizing: border-box;
+      transition: border-color 0.2s ease;
+    }
+
+    .form-input:focus {
+      outline: none;
+      border-color: var(--primary-color);
+      box-shadow: 0 0 0 3px rgba(16, 185, 129, 0.1);
+    }
+
+    .form-input:read-only {
+      background-color: #f9fafb;
+      color: #6b7280;
+    }
+  `
+
+  connectedCallback() {
+    super.connectedCallback()
+    this.updateComplete.then(() => {
+      const input = this.shadowRoot?.querySelector(
+        '.amount-input'
+      ) as HTMLInputElement
+      if (input) {
+        input.focus()
+      }
+    })
+  }
+
+  private getCurrencySymbol(): string {
+    const symbols: Record<string, string> = {
+      usd: '$',
+      eur: '€',
+      gbp: '£'
+    }
+    return symbols[this.walletAddress!.assetCode.toLowerCase()]
+  }
```

Comment:
Extract this into a shared util, instead of method. Can use the same one from extension.